### PR TITLE
Send feedback metadata to JaaS feedback endpoint

### PR DESCRIPTION
**Pull request by horymury**
I've changed the end of meeting promise resolves for room disconnect and send feedback from Promise.all to chained promises because I still need information from the room on sending the feedback metadata to JaaS endpoint, and with Promise.all, some of the information is purged on room disconnected before I get to store it for sending metadata.

### CODE REVIEW

File: react/features/feedback/logger.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// @flow
+
+import { getLogger } from '../base/logging/functions';
+
+export default getLogger('features/analytics');
```
Comment (quitrk) [inline react/features/feedback/logger.js:5]: should this be `features/feedback`?

File: react/features/feedback/actions.js [hunk 2]
```diff
@@ -110,6 +112,31 @@ export function openFeedbackDialog(conference: Object, onClose: ?Function) {
...
+/**
+ * Sends feedback metadata to JaaS endpoint.
+ *
+ * @param {Object} state - The redux state.
+ * @param {JitsiConference} conference - The JitsiConference that is being rated.
+* @param {Object} feedback - The feedback message and score.
+ *
+ * @returns {Promise}
+ */
+export function sendJaasFeedbackMetadata(state: Object, conference: Object, feedback: Object) {
```
Comment (quitrk) [inline react/features/feedback/actions.js:124]: maybe this should be an action?

File: react/features/feedback/functions.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Sends feedback metadata to JaaS endpoints.
+ *
+ * @param {string} url - The JaaS metadata endpoint URL.
+ * @param {Object} conference - The conference.
+ * @param {Object} feedbackData - The feedback data object.
+ * @returns {Promise}
+ */
+export async function sendFeedbackToJaaSRequest(url: string, conference: Object, feedbackData: Object) {
+    const { token } = conference.xmpp;
```
Comment (hristoterezov) [inline react/features/feedback/functions.js:16]: conference.xmpp  is not considered of the public API of lib-jitsi-meet. Anyway isn't the token and all the payload information already stored  in redux?

File: react/features/feedback/actions.js [hunk 4]
```diff
@@ -125,16 +154,17 @@ export function submitFeedback(
...
-                    error
-                });
-
-                return Promise.reject(error);
-            }
-        );
+    return (dispatch: Dispatch<any>) =>
+        conference.sendFeedback(score, message)
+        .then(() => dispatch({ type: SUBMIT_FEEDBACK_SUCCESS }))
+        .then(() => dispatch(sendJaasFeedbackMetadata(conference, { score,
```
Comment (hristoterezov) [inline react/features/feedback/actions.js:160]: Haven't we dispatched `sendJaasFeedbackMetadata` already trough  `conference.sendFeedback`?

File: conference.js [hunk 5]
```diff
@@ -1093,6 +1094,9 @@ export default {
      * @param detailedFeedback detailed feedback from the user. Not yet used
      */
     sendFeedback(overallFeedback, detailedFeedback) {
+        APP.store.dispatch(sendJaasFeedbackMetadata(room, { score: overallFeedback,
```
Comment (hristoterezov) [inline conference.js:1097]: Since this one is returning promise, maybe you want to wait for it to finish as well before resolving/rejecting the result from the function?

File: conference.js [hunk 6]
```diff
@@ -1093,6 +1094,9 @@ export default {
      * @param detailedFeedback detailed feedback from the user. Not yet used
      */
     sendFeedback(overallFeedback, detailedFeedback) {
```
Comment (hristoterezov) [inline conference.js:1096]: Optional: I don't see why this function needs to be in conference.js. It will be nice to move it out of conference.js. 

NOTE: This change is not related to your changes in the PR. But it should pretty easy to move it to react/redux.

**Review by hristoterezov — CHANGES_REQUESTED** [review]
left some comments!

File: conference.js (hunk 6 above)
Comment (horymury) [reply conference.js:1096]: I don't think this one is actually used anywhere. Think it's confusion of names, because in the feedback we call 'conference' the conference.room from conference.js. Basically conference in Feedback actions is the JitsiConference object from lib-jitsi-meet.

File: react/features/feedback/actions.js (hunk 4 above)
Comment (horymury) [reply react/features/feedback/actions.js:160]: As stated in my previous comment, think the conference.sendFeedback is not used anywhere. In the Feedback actions we call the .sendFeedback from conference.room ( JItsiConference instance) which calls the low level lib-jitsi-meet function.

File: conference.js (hunk 6 above)
Comment (horymury) [reply conference.js]: Removed this function completely as is does not seem to be used anywhere.

File: react/features/feedback/functions.js (hunk 3 above)
Comment (horymury) [reply react/features/feedback/functions.js]: changed to using the value from store

File: react/features/feedback/functions.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        sessionId,
+        meetingFqn,
+        score,
+        message
+    } = feedbackData;
+    const headers = {
+        'Authorization': `Bearer ${jwt}`,
+        'Content-Type': 'application/json'
+    };
+    const payload = jwtDecode(jwt);
```
Comment (hristoterezov) [inline react/features/feedback/functions.js:26]: Again, isn't the jwt already decoded and stored in redux?

**Review by hristoterezov — CHANGES_REQUESTED** [review]
left a comment

File: react/features/feedback/functions.js (hunk 7 above)
Comment (horymury) [reply react/features/feedback/functions.js]: yep, apparently tenant was not , but fixed it by checking context.tenant || sub || undefined.

**Review by hristoterezov — APPROVED** [review]
